myapp/src/services/facilityService.js

```diff
@@ -167,6 +165,14 @@ class FacilityService {
   }
 
   async updateFacility(id, data) {
+    const facility = await this.getFacilityById(id);
+    const status = await this.getFacilityRegistrationStatus(facility.authorId);
```

Comment:
This check will not allow facility update, if the facility is either
- made by admin without the request process, or any facilities that are pre-existent in DB
- request is deleted after being accepted